editoast/src/views/test_app.rs

```diff
@@ -158,6 +182,17 @@ where
             .expect("no DbConnectionPoolV2 setup")
             .clone()
     }
+
+    pub fn fetch(&self, req: Request) -> TestResponse {
+        futures::executor::block_on(async move {
+            tracing::debug!(target: "test_app", request = ?req, "Fetching test request");
+            let response = self.service.call(req).await.unwrap_or_else(|err| {
+                tracing::error!(target: "test_app", error = ?err, "Error fetching test request");
+                panic!("could not fetch test request");
+            });
+            TestResponse::new(response)
+        })
```

Comment:
These are logs, but maybe we'd prefer to open spans instead. Since you have an `async move` (not a function , so you cannot annotate with `#[tracing::instrument]`), you could use the `.instrument(some_span)` instead.

See https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code.

```suggestion
        futures::executor::block_on(async move {
            let response = self.service.call(req).await.unwrap_or_else(|err| {
                tracing::error!(error = ?err, "Error fetching test request");
                panic!("could not fetch test request");
            });
            TestResponse::new(response)
        }
        .instrument(tracing::debug_span!(target: "test_app", "fetching request", request = ?req))
```

Reply:
It's definitely better this way. I've made a few changes though:

1. I only include in the span the method and the url because otherwise the formatting of the log entries becomes weird (the request is printed over several lines).
2. Since we still want the whole debug representation of the request to be shown once, I still log it in the closure.

Reply:
Note that we could probably activate a pretty logger for tests so the "over several lines" is at least done right? Actually, even in production we could do that, logs are only intended to be consumed by humans, not machine since Opentelemetry will do the job for machines. And since one-line logs are only useful for machines, what about prettyfying everything.

https://docs.rs/tracing-subscriber/latest/tracing_subscriber/?search=pretty

Reply:
I agree. I'll do that in another PR so that if the change brings up some discussion that won't block this PR.

Reply:
https://github.com/OpenRailAssociation/osrd/pull/7548